gpt4all-chat/modellist.cpp

```diff
@@ -802,12 +802,22 @@ void ModelList::updateData(const QString &id, const QVector<QPair<int, QVariant>
             return;
         }
 
+        // We only sort when one of the fields used by the sorting algorithm actually changes that
+        // is implicated or used by the sorting algorithm
+        bool shouldSort = false;
+
         for (const auto &d : data) {
             const int role = d.first;
             const QVariant value = d.second;
             switch (role) {
             case IdRole:
-                info->setId(value.toString()); break;
+            {
+                if (info->id() != value.toString()) {
+                    info->setId(value.toString());
+                    shouldSort = true;
+                }
+                break;
+            }
```

Comment:
I would prefer not to have two levels of braces at the same level of indentation. You *could* just remove them as there aren't any variables here that need scoping.

If you prefer to have them then I would at least indent the block:
```suggestion
                {
                    if (info->id() != value.toString()) {
                        info->setId(value.toString());
                        shouldSort = true;
                    }
                    break;
                }
```